промежуточное ревью 1

Julyborn opened the pull request:
https://github.com/users/Julyborn/projects/1/views/1

Julyborn2 committed:
рефакторинг с разбиения по слоям на разбиение по фичам

Julyborn2 committed:
Настройка build.gradle и AndroidManifest.xml, добавление app и правка пакетов в фрагментах

Julyborn2 committed:
Исправление замечаний github-actions

Julyborn committed:
Merge pull request #2 from Julyborn/DIPLOMA-1

Diploma 1 (Настройка build.gradle и AndroidManifest.xml)

Myachkaa committed:
1. Определены цвета в colors.xml для светлой и темной темы согласно макету https://www.figma.com/design/gAWSqZ8mbc2g0oeuKPdL19/HH-(Android)?node-id=1-10363&node-type=frame&t=CseXYyF23uPa9Wpm-0
2. В themes.xml для светлой и темной темы определены два цвета(colorPrimary,colorOnPrimary), темы наследуются от Theme.MaterialComponents.DayNight.NoActionBar
3. Созданная тема приложения прописана в AndroidManifest.xml

Rai-gou committed:
New fragment_main

Rai-gou committed:
New fragment_favorites

Myachkaa committed:
Настройка Dependency Injection с использованием Koin
#7

1. Созданы базовые классы (заглушки) и интерфейсы для каждого слоя архитектуры

2. Созданы модули Koin для каждого слоя архитектуры (Data, Domain, Presentation).

3. Описана инициализация общих компонентов в соответствующих модулях.

4. Добавлена инициализация Koin в класс Application.

Myachkaa committed:
cleaning

Myachkaa committed:
Импорт шрифтов #10
1. Добавлены шрифты в папку font.
2. Описаны стили для текста в файле styles.xml, исходя из макета https://www.figma.com/design/gAWSqZ8mbc2g0oeuKPdL19/HH-(Android)?node-id=1-10356&node-type=frame&t=YOEFupq56PRVntEG-0
3. Размеры вынесены в файл dimens.xml

Julyborn committed:
Merge pull request #8 from Julyborn/New-Windows-Main-Favorites

New windows main favorites

Julyborn committed:
Merge pull request #6 from Julyborn/colors_and_team

Настройка темы

Julyborn committed:
Merge pull request #9 from Julyborn/DI_Koin_#7

Настройка Dependency Injection с использованием Koin #7

Julyborn committed:
Merge pull request #11 from Julyborn/fonts_#10

Импорт шрифтов #10

Julyborn2 committed:
Сверстал экран команды и добавил имена в strings.xml. Так же добавил в стили текстов цвета из тем тк на макетах они почти всегда одинаковые.
+поменял лейауты в FavoritesFragment и SearchFragment на заглушки

Julyborn2 committed:
Добавил новое имя

Vladi1989 committed:
Add app icon

Julyborn2 committed:
Взял остальные имена из пачки (так что без отчеств)

Rai-gou committed:
HeadHunterAPI

Rai-gou committed:
suspend HeadHunterAPI

Rai-gou committed:
DataClass SearchResponseVacancy

Julyborn committed:
Merge pull request #14 from Julyborn/DIPLOMA-5

Diploma 5

Rai-gou committed:
Merge branch 'refs/heads/New-Windows-Main-Favorites' into dev

Liocha committed:
Настройка графа навигации

Liocha committed:
Настройка графа навигации

Julyborn committed:
Merge pull request #17 from Julyborn/App_icon_14

App_icon_14

Julyborn committed:
Merge pull request #21 from Julyborn/20-bottom-navigation-setup

20 bottom navigation setup

Liocha committed:
Собрал базу/черновик для domain слоя по поиску вакансий.

Rai-gou committed:
DataClass Area

Rai-gou committed:
Space

Rai-gou committed:
Delete Space Again

Rai-gou committed:
Delete Space (

Rai-gou committed:
Merge remote-tracking branch 'origin/dev' into dev

Rai-gou committed:
Merge branch 'refs/heads/Retrofit' into dev

Rai-gou committed:
Employer

Rai-gou committed:
add Salary

Rai-gou committed:
SearchResponse

Rai-gou committed:
RetrofitInstance headHunterAPI

Julyborn2 committed:
Добавил функцию debounce в папку util
Создает функцию, которая ограничивает вызов [action] не чаще, чем раз в [delayMs] миллисекунд.
 *
delayMs Задержка в миллисекундах перед вызовом [action].
scope Корутинный Scope, в котором будет выполняться задержка.
action Функция, которая будет вызвана после задержки.
return Функция, принимающая параметр [T] и вызывающая [action] с задержкой.

Rai-gou committed:
NetworkUtils

Rai-gou committed:
object NetworkUtils

Julyborn2 committed:
Исправил замечания github-actions

Liocha committed:
Merge pull request #23 from Julyborn/22-search-query-domain-layer

Собрал базу/черновик для domain слоя по поиску вакансий.

Julyborn2 committed:
Исправил замечания github-actions 2

Julyborn committed:
Merge pull request #27 from Julyborn/24------debounce

Добавил функцию debounce в папку util

Liocha committed:
Собрал черновик для presentation слоя

Liocha committed:
detekt fix warnings

Liocha committed:
detekt fix warnings

Liocha committed:
detekt fix warnings

Rai-gou committed:
Merge remote-tracking branch 'origin/dev' into dev

Liocha committed:
detekt fix warnings

Rai-gou committed:
Merge branch 'refs/heads/dev' into Prepare-data-layer

Liocha committed:
detekt fix warnings

Liocha committed:
detekt fix warnings

Liocha committed:
Merge pull request #30 from Julyborn/29-job-search-screen-presentation-layer

Собрал черновик для presentation слоя

Rai-gou committed:
into DTO

Rai-gou committed:
add Repository

Rai-gou committed:
change @GET

Julyborn2 committed:
Функция форматирования модели Salary в строку String

Rai-gou committed:
add catch error

Rai-gou committed:
add get()

Rai-gou committed:
change SearchResponse

Rai-gou committed:
change BillingType

Rai-gou committed:
Merge remote-tracking branch 'origin/dev' into dev

Rai-gou committed:
Merge branch 'refs/heads/Prepare-data-layer' into dev

# Conflicts:
#	app/src/main/java/ru/practicum/android/diploma/search/domain/models/Vacancy.kt

Vladi1989 committed:
Add fragment_search

Liocha committed:
Merge branch 'dev' into Prepare-data-layer

Liocha committed:
Merge pull request #26 from Julyborn/Prepare-data-layer

Prepare data layer

Myachkaa committed:
Merge pull request #34 from Julyborn/32-salary-to-string

Функция форматирования модели Salary в строку String

Myachkaa committed:
Верстка экрана "Детали вакансии" №33

Liocha committed:
Draft implementation search request

Liocha committed:
Merge pull request #38 from Julyborn/36-search-query

Draft implementation search request

Rai-gou committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	app/src/main/java/ru/practicum/android/diploma/search/domain/models/Vacancy.kt

Vladi1989 committed:
Merge pull request #35 from Julyborn/Search_screen_#28

Task #28

Rai-gou committed:
Merge remote-tracking branch 'origin/dev' into dev

Liocha committed:
fix assemble job

Liocha committed:
fix detect warn

Liocha committed:
Merge pull request #40 from Julyborn/fix-checkers

Fix checkers

AleksandrIlinskii commented:
Обратите внимание на работу с ветками, в соответствии с заданием 
<img width="745" alt="image" src="https://github.com/user-attachments/assets/c1e3b4df-b785-4d33-b3df-10c203734503">

AleksandrIlinskii commented:
При старте приложение должен быть плейсхолдер, а не плашка что "ничего не найдено"

![telegram-cloud-photo-size-4-6048501683480282768-y](https://github.com/user-attachments/assets/77dedcdc-ff1f-4e48-8ab9-bd9ce5997d6a)

AleksandrIlinskii commented:
Темная тема не поддерживается

AleksandrIlinskii commented on .idea/.name:
Папку idea стоит добавить в gitignore, тут лишние файлы для пуша в репозиторий

AleksandrIlinskii commented on app/build.gradle.kts:
```diff
@@ -36,17 +37,37 @@ android {
 
 dependencies {
+    val navVersion = "2.8.1"
+    val fragmentVersion = "1.8.3"
```
Все версии библиотек должны быть в файле libs.version.toml

AleksandrIlinskii commented on app/build.gradle.kts:
```diff
@@ -55,4 +76,9 @@ dependencies {
     // endregion
+
+    // Navigation
+    implementation("androidx.navigation:navigation-fragment:$navVersion")
+    implementation("androidx.navigation:navigation-ui:$navVersion")
+    implementation("androidx.fragment:fragment-ktx:$fragmentVersion")
```
Библиотеки нужно добавить в файл libs.version.toml

AleksandrIlinskii commented on app/build.gradle.kts:
```diff
@@ -36,17 +37,37 @@ android {
+    kapt(libs.room.compiler)
+    implementation(libs.room.ktx)
+
     // region Unit tests
```
По традиции все зависимости, связанные с тестами, должны быть в конце

AleksandrIlinskii commented on app/src/main/AndroidManifest.xml:
```diff
@@ -2,19 +2,25 @@
     xmlns:tools="http://schemas.android.com/tools">
 
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+    <uses-permission android:name="android.permission.INTERNET" />
```
Между permission и следующим блоком стоит добавить пустую строку

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/favorites/ui/FavoritesFragment.kt:
```diff
@@ -0,0 +1,19 @@
+    ): View? {
+        return inflater.inflate(R.layout.fragment_favorites, container, false)
+    }
+}
```
Binding следует добавить в каждый фрагмент согласно документации Google
https://developer.android.com/topic/libraries/view-binding?hl=ru#fragments

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/root/RootActivity.kt:
```diff
@@ -0,0 +1,41 @@
+
+        navController.addOnDestinationChangedListener { _, destination, _ ->
+            when (destination.id) {
+                R.id.filterFragment -> {
+                    bottomNavigationView.visibility = View.GONE
+                }
+
+                R.id.vacanciesFragment -> {
+                    bottomNavigationView.visibility = View.GONE
+                }
```
```suggestion
                R.id.filterFragment, 
                R.id.vacanciesFragment -> bottomNavigationView.visibility = View.GONE
```

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/search/data/impl/SearchRepositoryImpl.kt:
```diff
@@ -0,0 +1,100 @@
+import ru.practicum.android.diploma.search.domain.models.VacancySearchParams
+import ru.practicum.android.diploma.search.domain.models.toMap
+import java.io.IOException
+
+class SearchRepositoryImpl(
+    private val api: HeadHunterAPI,
+    private val context: Context
+) : SearchRepository {
+
+    @RequiresExtension(extension = Build.VERSION_CODES.S, version = 7)
```
В какой момент времени понадобилась эта аннотация?

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/search/data/impl/SearchRepositoryImpl.kt:
```diff
@@ -0,0 +1,100 @@
+        )
+    }
+
+    private fun getArea(areaDto: AreaDto?): String {
+        if (areaDto == null) {
+            return ""
+        }
+        return areaDto.name ?: ""
+    }
```
```suggestion
    private fun getArea(areaDto: AreaDto?): String =
       areaDto?.name.orEmpty()
```

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/search/data/network/HeadHunterAPI.kt:
```diff
@@ -0,0 +1,21 @@
+    // Поиск вакансий
+    @GET("vacancies")
+    suspend fun searchVacancies(
+        @Header("Authorization") authToken: String?,
+        @Header("User-Agent") userAgent: String?,
```
Если вы добавите интерсептор - вам не придется добавлять header для каждого запроса

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/search/presentation/SearchViewModel.kt:
```diff
@@ -0,0 +1,56 @@
+
+class SearchViewModel(val searchInteractor: SearchInteractor) : ViewModel() {
+    private val _uiState = MutableLiveData<UiScreenState>(UiScreenState.Default)
+    val uiState: LiveData<UiScreenState> get() = _uiState
```
```suggestion
    val uiState: LiveData<UiScreenState> 
        get() = _uiState
```

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/search/presentation/SearchViewModel.kt:
```diff
@@ -0,0 +1,56 @@
+        when (result) {
+            is Resource.NoInternetError -> _uiState.value = UiScreenState.NoInternetError
+            is Resource.ServerError -> _uiState.value = UiScreenState.ServerError
+            is Resource.Success -> if (result.data.isEmpty()) {
+                _uiState.value = UiScreenState.Empty
+            } else {
+                val vacanciesUi = result.data.map { vacancy ->
+                    mappingVacancyToVacancyUi(vacancy)
+                }
+                _vacanciesList.value = _vacanciesList.value?.plus(vacanciesUi)
+                _uiState.value = UiScreenState.Success(vacancies = vacanciesUi, found = result.found ?: 0)
+            }
```
Отступы неккоректные, нужно сделать code - reformat code

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/search/ui/SearchFragment.kt:
```diff
@@ -0,0 +1,80 @@
+import ru.practicum.android.diploma.search.presentation.models.VacancyUi
+
+class SearchFragment : Fragment() {
+    private var binding: FragmentSearchBinding? = null
```
Binding для фрагмента сделан некорректно

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/search/ui/SearchFragment.kt:
```diff
@@ -0,0 +1,80 @@
+    }
+
+    @Suppress("EmptyBlock")
+    private fun showDefaultState() {
+        // Implement this method later
+    }
```
Вместо того, чтобы suppress'ом затыкать детект
```suggestion
    private fun showDefaultState() = Unit
```

AleksandrIlinskii commented on app/src/main/java/ru/practicum/android/diploma/util/Utils.kt:
```diff
@@ -0,0 +1,25 @@
+    val fromSalary = this.from
+    val toSalary = this.to
+    val currencySymbol = when (this.currency) {
+        "RUB" -> "₽"
+        "USD" -> "$"
+        "EUR" -> "€"
```
3 валюты - очень мало, у нас очень много вакансий из других стран (спойлер, не только RUB код для рубля)

AleksandrIlinskii commented on app/src/main/res/layout/fragment_favorites.xml:
```diff
@@ -0,0 +1,17 @@
+             android:layout_height="match_parent">
+
+    <TextView
+        android:id="@+id/textView2"
```
textView не лучшее название

AleksandrIlinskii commented on app/src/main/res/layout/fragment_search.xml:
```diff
@@ -0,0 +1,211 @@
+        android:background="@drawable/img_search"
+        android:layout_marginHorizontal="16dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintDimensionRatio="328:223"
```
Какие-то очень странные цифры у вас тут

AleksandrIlinskii commented on app/src/main/res/layout/fragment_team.xml:
```diff
@@ -0,0 +1,77 @@
+        android:text="@string/member5"
+        style="@style/TextMedium16"/>
+
+</LinearLayout>
+</LinearLayout>
```
Нужно поправить отступы

AleksandrIlinskii commented on gradle/libs.versions.toml:
```diff
@@ -14,6 +18,14 @@ detektTwitterComposeRules ="0.0.26"
 [libraries]
 
 # Detekt
+compiler = { module = "com.github.bumptech.glide:compiler", version.ref = "glide" }
```
Лучше назвать это glide-compiler

Liocha committed:
draft by search query

Liocha committed:
Merge pull request #42 from Julyborn/search-query-draft

draft by search query

Myachkaa committed:
Merge branch 'dev' of https://github.com/Julyborn/practicum-android-diploma into details_data_domain_41

# Conflicts:
#	app/src/main/res/values/dimens.xml
#	app/src/main/res/values/strings.xml
#	app/src/main/res/values/styles.xml

Rai-gou committed:
Merge remote-tracking branch 'origin/dev' into dev

Rai-gou committed:
add Toast

Julyborn2 committed:
Исправил ошибку с отображением рубля в функцию formatSalary

Myachkaa committed:
Эпик 2

Реализованы задачи:
- Верстка экрана «Детали вакансии» #33
- Проработка слоев Data и Domain для экрана Детали вакансии #41
- Проработка слоев Presentation и UI для экрана Детали вакансии #44

Myachkaa committed:
cleaning

Julyborn2 committed:
Добавил пагинацию
Подправил отображение recyclerview (нижняя граница выходила за пределы экрана)

Vladi1989 committed:
Add fragment_search

Julyborn2 committed:
Добавил в debounce возможность отмены .cancel
Добавил возможность поиска по кнопке ввода на клавиатуре

Rai-gou committed:
change getArea

Rai-gou committed:
add AuthInterceptor

Rai-gou committed:
change AuthInterceptor

Rai-gou committed:
change uiState

Rai-gou committed:
add more currencySymbol

Julyborn committed:
Merge pull request #46 from Julyborn/31-paging

Пагинация

Julyborn committed:
Merge branch 'dev' into Search_screen_#28

Julyborn committed:
Merge pull request #48 from Julyborn/Search_screen_#28

Search screen #28

Myachkaa committed:
Merge branch 'dev' of https://github.com/Julyborn/practicum-android-diploma into vacancy_details_37_41_44

# Conflicts:
#	app/src/main/res/layout/fragment_search.xml

Myachkaa committed:
merge

Myachkaa committed:
Merge pull request #45 from Julyborn/vacancy_details_37_41_44

Эпик 2

Myachkaa committed:
Добавлен плейсхолдер "Ошибка сервера" на экран "Детали вакансии"

Myachkaa committed:
cleaning

Myachkaa committed:
Merge pull request #50 from Julyborn/vacancy_details_37_41_44

Добавлен плейсхолдер "Ошибка сервера" на экран "Детали вакансии"

Rai-gou committed:
add bild

Rai-gou committed:
Merge branch 'dev' into правки

Rai-gou committed:
Merge pull request #47 from Julyborn/правки

Правки по замечаниям

Rai-gou committed:
update Strings

Rai-gou committed:
update SerchFragment

Rai-gou committed:
Merge pull request #53 from Julyborn/правки-повтор

правки проверка

Rai-gou committed:
update DebounceUtils, FavoritesFragment

Rai-gou committed:
update SearchRepositoryImpl

Rai-gou committed:
update VacanciesViewModel

Rai-gou committed:
Merge pull request #55 from Julyborn/правки-повтор

правки повтор2

Liocha committed:
added favorites vacancies

Liocha committed:
Merge pull request #58 from Julyborn/52-favorite-vacancies

added favorites vacancies

Julyborn2 committed:
Исправил ошибки

Julyborn2 committed:
Добавил кнопку очистки поля ввода поиска, когда оно не пустое

Julyborn2 committed:
Убрал варнинги

Julyborn2 committed:
Убрал варнинги 2

Julyborn2 committed:
Убрал варнинги 3

Julyborn committed:
Merge pull request #60 from Julyborn/59-DarkThemeColorsFix

Исправил ошибки
